OneSchedule.Tests/OnesheduleStateTests.cs

```diff
@@ -0,0 +1,225 @@
+using Moq;
+using NUnit.Framework;
+using OneSchedule.Data.Abstractions;
+using OneSchedule.Domain.Abstractions.StateMachine;
+using OneSchedule.Domain.Models;
+using OneSchedule.Domain.StateMachine.AskState;
+using OneSchedule.Domain.StateMachine.GetState;
+using OneSchedule.Entities;
+using OneSchedule.Exceptions.CustomExceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Telegram.Bot;
+using Telegram.Bot.Types.Enums;
+using Telegram.Bot.Types.ReplyMarkups;
+
+namespace OneSchedule.Tests
+{
+    public class OnesheduleStateTests
+    {
+        private Mock<ITelegramBotClient> _botMoq;
+        private readonly Mock<IStateContext> _stateContextMoq;
+
+        public OnesheduleStateTests()
+        {
+            _botMoq = new Mock<ITelegramBotClient>();
+            _stateContextMoq = new Mock<IStateContext>();
+        }
+
+        [Test]
+        public async Task AskBeginDateState_IsCompleted()
```

Comment:
The name of tests usually contains three parts: name of the method you are testing, input data or short description of the particular scenario, result or output of test. Sometimes, for general happy path you can omit the second part of the name. So, this test you can be named like this AskBeginDateState_Success or AskBeginDateState_HappyPath